payments-core/src/main/java/com/stripe/android/IntentConfirmationInterceptor.kt

```diff
@@ -93,30 +93,40 @@ class DefaultIntentConfirmationInterceptor @Inject constructor(
                 setupForFutureUsage = setupForFutureUsage,
             )
         } else {
-            createPaymentMethod(
-                paymentMethodCreateParams.copy(
-                    productUsage = paymentMethodCreateParams.attribution
-                        .plus("deferred-intent")
-                )
-            ).fold(
-                onSuccess = { paymentMethod ->
-                    intercept(
-                        clientSecret = null,
-                        paymentMethod = paymentMethod,
-                        shippingValues = shippingValues,
-                        setupForFutureUsage = setupForFutureUsage,
-                    )
-                },
-                onFailure = { error ->
-                    NextStep.Fail(
-                        cause = error,
-                        message = genericErrorMessage,
-                    )
-                }
+            handleDeferredIntent(
+                shippingValues = shippingValues,
+                paymentMethodCreateParams = paymentMethodCreateParams,
+                setupForFutureUsage = setupForFutureUsage,
             )
         }
     }
 
+    private suspend fun handleDeferredIntent(
+        paymentMethodCreateParams: PaymentMethodCreateParams,
+        shippingValues: ConfirmPaymentIntentParams.Shipping?,
+        setupForFutureUsage: ConfirmPaymentIntentParams.SetupFutureUsage?,
+    ): NextStep {
+        val params = paymentMethodCreateParams.copy(
+            productUsage = paymentMethodCreateParams.attribution + "deferred-intent",
```

Comment:
Do we have tests for this?

Reply:
Yep, this is tested in `PaymentSheetTest` and `FlowControllerTest` via the `NetworkRule`.